### PR TITLE
Failing Foundation.Date roundtrip coding test

### PR DESCRIPTION
**Pull request by kirilltitov**
Reproduces #6.

### CODE REVIEW

**Comment by kirilltitov** [comment]
Fails on macOS, haven't checked Linux.

**Comment by kirilltitov** [comment]
It fails on my Linux setup:

```
root@caad6a67f690:/tmp/MessagePack# swift test
Compile Swift Module 'MessagePack' (11 sources)
/tmp/MessagePack/Sources/MessagePack/Decoder/SingleValueDecodingContainer.swift:168:73: error: use of unresolved identifier 'NSEC_PER_SEC'; did you mean 'CLOCKS_PER_SEC'?
        let timeInterval = TimeInterval(seconds) + nanoseconds / Double(NSEC_PER_SEC)
                                                                        ^~~~~~~~~~~~
                                                                        CLOCKS_PER_SEC
SwiftGlibc.CLOCKS_PER_SEC:1:12: note: 'CLOCKS_PER_SEC' declared here
public var CLOCKS_PER_SEC: clock_t { get }
           ^
/tmp/MessagePack/Sources/MessagePack/Encoder/SingleValueEncodingContainer.swift:204:54: error: use of unresolved identifier 'NSEC_PER_SEC'; did you mean 'CLOCKS_PER_SEC'?
        let nanoseconds = UInt32(fractional * Double(NSEC_PER_SEC))
                                                     ^~~~~~~~~~~~
                                                     CLOCKS_PER_SEC
SwiftGlibc.CLOCKS_PER_SEC:1:12: note: 'CLOCKS_PER_SEC' declared here
public var CLOCKS_PER_SEC: clock_t { get }
           ^
error: terminated(1): /usr/bin/swift-build-tool -f /tmp/MessagePack/.build/debug.yaml test output:
```

I think we have another bug here :(

**Comment by kirilltitov** [comment]
For some reason `NSEC_PER_SEC` const isn't available in Linux Foundation, so as a temporary workaround I added it myself. Now tests are correctly invoked, and my test _properly_ fails :)

File: Sources/MessagePack/Decoder/SingleValueDecodingContainer.swift [hunk 1]
```diff
@@ -1,5 +1,9 @@
 import Foundation
 
+#if os(Linux)
+let NSEC_PER_SEC: UInt64 = 1000000000
```
Comment (kirilltitov) [inline Sources/MessagePack/Decoder/SingleValueDecodingContainer.swift]: I wouldn't rely on original Foundation const as it's not available in Linux Foundation (for reason unknown).

Second thought: you should add another Travis check, on Linux environment.

**Comment by mattt** [comment]
Following up on this, the failing test is now passing thanks to 5c56210 and 6c7161b. Thanks again for your help on this, @kirilltitov!

**Pull request closed without merging**